enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -626,22 +608,57 @@ def can_redeem(self, request, enterprise_customer_uuid):
                         "reason": reason,
                         "policy_uuids": [policy.uuid for policy in policies],
                     })
+
+            # Select one redeemable policy (i.e. use the "policy resolver") and serialize it for output.
+            resolved_policy = None
             if redeemable_policies:
                 resolved_policy = SubsidyAccessPolicy.resolve_policy(redeemable_policies)
-                serialized_policy = serializers.SubsidyAccessPolicyRedeemableResponseSerializer(resolved_policy).data
 
+            # Determine if the learner has already redeemed the requested content_key.
             has_successful_redemption = any(
                 redemption['state'] == TransactionStateChoices.COMMITTED
                 for redemption in redemptions
             )
-            can_redeem_for_content_response = {
+
+            # Determine the price for content for display purposes only.
+            try:
+                content_metadata = self.subsidy_client.get_subsidy_content_data(enterprise_customer_uuid, content_key)
+                # Note that the "content_price" key is guaranteed to exist, but the value may be None.  Also, the
+                # upstream field type is a string representing a decimal dollar.
+                list_price_decimal_dollars_str = content_metadata["content_price"]
+                if list_price_decimal_dollars_str:
+                    list_price_decimal_dollars = float(list_price_decimal_dollars_str)
+                    list_price_integer_cents = int(list_price_decimal_dollars * 100)
+                else:
+                    list_price_decimal_dollars = None
+                    list_price_integer_cents = None
+            except requests.exceptions.HTTPError:
+                # No need to record a failure reason here because evaluate_policies() -> policy.can_redeem() already
+                # caught any problematic or non-existent content keys, and provided a reason it could not be redeemed.
+                # Just set a None content_price and move along.
+                list_price_decimal_dollars = None
+                list_price_integer_cents = None
```

Comment:
This method body is getting really long - is there an obvious place to wrap this in a helper method or something?  Or maybe the serializer can do the translation of string to numbers?

Reply:
I'd love to consolidate this into the serializer, but in light of my long todo list I'm going to defer this and add a comment.